haddock-api/src/Haddock/Types.hs

```diff
@@ -280,7 +284,16 @@ data DocName
   | Undocumented Name
      -- ^ This thing is not part of the (existing or resulting)
      -- documentation, as far as Haddock knows.
-  deriving Eq
+  deriving (Eq, Data)
+
+type instance PostRn DocName NameSet  = PlaceHolder
```

Comment:
Previously it was defined in `Haddock.Interface.Rename` module (not sure what for) but it was needed elsewhere, so I moved it to `Haddock.Types`. And basically, it makes `DocName` an instance of [`DataId`](https://downloads.haskell.org/~ghc/7.10.1/docs/html/libraries/ghc-7.10.1/PlaceHolder.html#t:DataId) which is required to use SYB stuff with `HsType`.